src/views/Ifos/hooks/useIfoApprove.ts

```diff
@@ -1,13 +1,27 @@
 import { useCallback } from 'react'
 import { ethers, Contract } from 'ethers'
 import { useCallWithGasPrice } from 'hooks/useCallWithGasPrice'
+import { TransactionResponse, TransactionReceipt } from '@ethersproject/providers'
 
 const useIfoApprove = (tokenContract: Contract, spenderAddress: string) => {
   const { callWithGasPrice } = useCallWithGasPrice()
-  const onApprove = useCallback(async (): Promise<ethers.providers.TransactionReceipt> => {
-    const tx = await callWithGasPrice(tokenContract, 'approve', [spenderAddress, ethers.constants.MaxUint256])
-    return tx.wait()
-  }, [spenderAddress, tokenContract, callWithGasPrice])
+  const onApprove = useCallback(
+    async (
+      onTransactionSubmitted: (tx: TransactionResponse) => void,
+      onSuccess: (receipt: TransactionReceipt) => void,
+      onError: (receipt: TransactionReceipt) => void,
+    ) => {
+      const tx = await callWithGasPrice(tokenContract, 'approve', [spenderAddress, ethers.constants.MaxUint256])
+      onTransactionSubmitted(tx)
+      const receipt = await tx.wait()
+      if (receipt.status) {
```

Comment:
Can't this be error status code and still be truthy?

Reply:
According to this (https://eips.ethereum.org/EIPS/eip-1066) 0 comes only when there is failure